google/cloud/storage/bucket.py

```diff
@@ -2361,7 +2361,9 @@ def lifecycle_rules(self):
             elif action_type == "SetStorageClass":
                 yield LifecycleRuleSetStorageClass.from_api_repr(rule)
             else:
-                raise ValueError("Unknown lifecycle rule: {}".format(rule))
+                warnings.warn(
```

Comment:
Does it cause issues if nothing is yielded here?

Reply:
Let me double check on that, thanks! Any insight on this @andrewsg?

Reply:
I considered that in my review but decided it looked to me like because it's in a for loop, yielding nothing will behave identically to no rule at all, which is probably acceptable in this case. I'm not familiar with how people use this feature, however.

Reply:
Gotcha, I guess it will just continue to the next rule in the list and/or end the generator if nothing is left.

I was discussing this issue with someone who pointed me to this code in the Java Bigtable client: https://github.com/googleapis/java-bigtable/blob/master/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/EncryptionInfo.java#L48 . I like the message there-- could we also have the warning suggest upgrading to a newer client version?

Reply:
To the point of future proof, I think suggesting upgrading to a newer client version would make sense. Will incorporate that in!